### PR TITLE
CI: troubleshoot py310 build

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/algorithms.py [hunk 1]
```diff
@@ -1092,18 +1093,19 @@ def checked_add_with_arr(
     # it is negative, we then check whether its sum with the element in
     # 'arr' exceeds np.iinfo(np.int64).min. If so, we have an overflow
     # error as well.
+    i8max = Timestamp.max.value  # GH#?
```
Comment (jreback) [inline pandas/core/algorithms.py:1096]: can u make these consistent across all of the files

iow this is called different names but the same even in this change set

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/algorithms.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/algorithms.py:1096]: will standardize in next commit.  this kludge really shouldn't be necessary

**Comment by lithomas1** [comment]
Hey @jbrockmendel Just giving the heads up that the Python 3.10 dev build was actually "fake" green and can be green even if failing. I'm trying to address that in #41595, so I guess you can ignore failures listed on the issue #41940. 

Also, FYI the Python dev build on tests on a single core, so you might find it helpful to change add an ``-n auto`` to the Pytest command. If you want to have CI fail on a failure, you can just remove ``continue-on-error`` flag.

**Comment by jbrockmendel** [comment]
Thanks @lithomas1.  IIUC what we really want is allow-failure which GH actions doesn't support (https://github.com/actions/toolkit/issues/399).  For now I'll be happy if non-breaking PRs get a little green checkmark.

**Comment by simonjayhawkins** [comment]
@jbrockmendel there's some mypy failures

**Comment by fangchenli** [comment]
> Also, FYI the Python dev build on tests on a single core, so you might find it helpful to change add an `-n auto` to the Pytest command.

The last time I tried, only single core works. We probably should try pytest-xdist master now.

**Comment by jbrockmendel** [comment]
Remaining failures look unrelated

File: pandas/core/algorithms.py (hunk 1 above)
Comment (jreback) [reply pandas/core/algorithms.py]: ok ping when ready.

Comment (jbrockmendel) [reply pandas/core/algorithms.py]: ping, will be nice to finally get back to green

**Review by jreback — APPROVED** [review]

File: pandas/core/algorithms.py (hunk 1 above)
Comment (jreback) [reply pandas/core/algorithms.py]: right what i mean was can you standardize these names (you mostly do, but some are CAPITAL and some not). prefer to do it in this PR as going to backport.

Comment (jbrockmendel) [reply pandas/core/algorithms.py]: the ones in .py files are all `i8max`, `u8max`, or `iNaT`

File: pandas/core/sorting.py [hunk 2]
```diff
@@ -40,7 +40,7 @@
     from pandas import MultiIndex
     from pandas.core.indexes.base import Index
 
-_INT64_MAX = np.iinfo(np.int64).max
+_INT64_MAX = lib.i8max
```
Comment (jreback) [inline pandas/core/sorting.py:43]: can u fix these rather than just renaming

Comment (jreback) [reply pandas/core/sorting.py:43]: alt change the ones in the .pyx to be uppercase 

it's not at all clear that these are actually the same

Comment (jbrockmendel) [reply pandas/core/sorting.py]: Updated+green

File: pandas/_libs/lib.pyi [hunk 3]
```diff
@@ -25,6 +25,9 @@ class NoDefault(Enum): ...
 
 no_default: NoDefault
 
+i8max: int
```
Comment (jreback) [inline pandas/_libs/lib.pyi:28]: i still find this odd, pls do in a followup (e.g. change these to the same naming as in util e.t. INT64_MAX)

**Comment by jreback** [comment]
tell you what. i like green, so if you can do a followup for the name cleaning (which we can backport to 1.3)

**Comment by jreback** [comment]
@meeseeksdev backport 1.3.x

**Pull request merged**